### ISSUE TITLE
Resource does not seem to differentiate between branch and tag

### ISSUE THREAD

**Issue by pivotal-pmital** [concourse/git-resource#330]
### Issue description
We have a pipeline configured to `put` to one of our specific repo's branch. However, the `put` task in our job started failing with errors shown in the screenshot attached. Our resource was pointing to `branch: passed` so this error didn't make sense to us as the latest commit on this branch was only 4 days old but the task shows that it's 777 commits behind. 

<img width="1422" alt="Screen Shot 2020-08-10 at 2 39 02 PM" src="https://user-images.githubusercontent.com/39704504/89818360-9fcd8b00-db17-11ea-923f-b02c49a17331.png">

When we looked into our concerned repo, we found that we also had a `tag` with the same name and for some reason, the resource was picking up the tag and not the branch. Hence, we were getting this error. We deleted the tag from our repo and the task started passing again.

<hr>

### Steps to reproduce the issue
1. Create a 'passed' branch a few commits behind the default/master branch
2. Create a 'passed' tag attached to a commit much older than the 'passed' branch
3. Spin up a pipeline that `puts` the 'passed' with all the new commits from the master branch.

<hr>

### What's the expected result?
- The pipeline puts to the `passed` branch.

<hr>

### What's the actual result?
- The pipeline tries to put to the `tag` instead

### Additional Note
We only noticed this when we upgraded from our concourse `5.7.1` to `6.4.1`

**Comment by taylorsilva** [comment]
I don't think this is something we can fix. I think the issue is `git` itself can't differentiate between the tag or branch. You can see the confusion this causes from this SO question: https://stackoverflow.com/questions/32609195/git-tags-with-the-same-name-with-a-branch

TLDR: don't have branches and tags with the same name; git won't know which one you're referring to and appears to prefer tags over branches when finding the relevant ref.
